main.go

```diff
@@ -133,10 +152,37 @@ func (s *admissionWebhookServer) unmarshal(in *admissionv1.AdmissionRequest) (p
 		return "", nil, nil
 	}
 	p = path.Join("/", p)
-	if metaPtr.Labels == nil {
-		metaPtr.Labels = make(map[string]string)
+	if podMetaPtr.Labels == nil {
+		podMetaPtr.Labels = make(map[string]string)
+	}
+	// Annotations shouldn't be applied second time.
+	if isReplicaOwnedByDeployment(in.Kind.Kind, metaPtr) {
+		return "", nil, nil
+	}
+	updatePodAnnotations(in.Kind.Kind, metaPtr, podMetaPtr)
+
+	return p, podMetaPtr, podSpec
+}
+
+func isReplicaOwnedByDeployment(kind string, metaPtr *v1.ObjectMeta) bool {
+	if kind == replicaSetKind {
+		for _, o := range metaPtr.OwnerReferences {
+			if o.Kind == deploymentKind {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func updatePodAnnotations(kind string, metaPtr, podMetaPtr *v1.ObjectMeta) {
```

Comment:
Why we cant do this directly in pod branch in the switch case above?

Reply:
Because the method is already big enough and lint raise errors when I add more code there.

Reply:
Can we fix the linter?

Reply:
I used inline function as you proposed and it worked, so there is no need to change linter settings.